### PR TITLE
Fix the error that is wrong link title of a downloadable product when enabling "Use Default Value"

### PR DESCRIPTION
**Pull request by tna274**
### Description (*)

Fix the error that is wrong link title of a downloadable product when enabling "Use Default Value"

### Related Pull Requests

N/A

### Fixed Issues (if relevant)

#27169 - Not able to update value with "use default checkbox" for Downloadable Product

### Manual testing scenarios (*)

1. Create a new downloadable product with a link
2. Choose scope at any store view (https://prnt.sc/rgpgs4)
3. In "Downloadable Information" section, uncheck "Use Default Value" box under link title and change link title then saving product
4. At the same product and scope, enable (check) "Use Default Value" box under link title then saving product

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Downloadable/Model/Link/Builder.php [hunk 1]
```diff
@@ -132,6 +132,10 @@ public function build(\Magento\Downloadable\Api\Data\LinkInterface $link)
         if (isset($this->data['is_unlimited']) && $this->data['is_unlimited']) {
             $link->setNumberOfDownloads(0);
         }
+
+        if (isset($this->data['use_default_title']) && $this->data['use_default_title'] == '1') {
```
Comment (eduard13) [inline app/code/Magento/Downloadable/Model/Link/Builder.php]: What do you think about applying the following minor adjustment?
```suggestion
        $useDefaultTitle = $this->data['use_default_title'] ?? false;
        
        if ($useDefaultTitle) {
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @tna274, thank you for your contribution, could you please review the minor recommendation?
Could you please also check and fix the failing Static Tests?
Thank you.

File: app/code/Magento/Downloadable/Model/Link/Builder.php (hunk 1 above)
Comment (tna274) [reply app/code/Magento/Downloadable/Model/Link/Builder.php]: Thanks for your idea.
I will apply your solution, it makes my code neater

**Comment by tna274** [comment]
Hi @eduard13 
Could you help me check the functional test. I found some failure test in report but they do not relate to my edited code.
The uncompleted tests relate to cms_block_page and simple product on frontend.

**Review by eduard13 — DISMISSED** [review]
Indeed, the failing tests aren't related to these changes.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-7137](https://jira.corp.magento.com/browse/ENGCOM-7137) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@tna274 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi, @tna274.

During testing, we faced the issue.

**_Problem:_** Not able to update the **Samples Title** by "**Use Default Value**" checkbox

**Manual testing scenario:**

1. Create a new downloadable product with a link and sample;
2. Save the product;
screen1
3. Choose scope at any store view;
screenshot2
4. In "Downloadable Information" section, uncheck "**Use Default Value**" boxes under **link** title and **sample** title, and change the values.
5. Save the product;
screenshot3
6. Now, check "**Use Default Value**" boxes under the link and sample titles;
7. Save the product;

**Actual Result:**
:heavy_multiplication_x:  The default value for the **link title** was saved but for the **sample title** was not.
screenshot4

@tna274 Could you take a look?

Thanks!

**Comment by tna274** [comment]
Hi @engcom-Alfa 
I did not know that case
OK, I will update it immediately

**Comment by tna274** [comment]
Hi @engcom-Alfa
I've been update PR, please check it again.

**Comment by engcom-Alfa** [comment]
Hi,@eduard13. Could you review the latest changes? Thanks!

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-7137](https://jira.corp.magento.com/browse/ENGCOM-7137) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Kilo** [comment]
@magento run Functional Tests B2B, @magento run Functional Tests CE, @magento run Functional Tests EE

**Comment by engcom-Echo** [comment]
@magento run all tests

**Pull request merged**